consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -0,0 +1,765 @@
+        mock_block.header.state_root_hash = '6'
+        mock_block.header.batch_ids = '4'
+
+        # create mock_signup_info
+        mock_signup_info.create_signup_info.return_value = \
+            mock.Mock(
+                poet_public_key='poet public key',
+                proof_data='proof data',
+                anti_sybil_id='anti-sybil ID',
+                sealed_signup_data='sealed signup data')
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                        'LOGGER') as mock_logger:
+            block_publisher = \
+                poet_block_publisher.PoetBlockPublisher(
+                    block_cache=mock_block_cache,
+                    state_view_factory=mock_state_view_factory,
+                    batch_publisher=mock_batch_publisher,
+                    data_dir=self._temp_dir,
+                    validator_id='validator_deadbeef')
+
+            self.assertFalse(
+                block_publisher.initialize_block(
+                    block_header=mock_block.header))
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.debug.call_args
+            self.assertTrue('No public key found, so going '
+                            'to register new signup information' in message)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'SignupInfo')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'PoetKeyStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.utils')
+    def test_no_validator_registry(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_key_state_store,
+            mock_signup_info,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ Test verifies that PoET Block Publisher fails if
+        a block is claimed by an unknown validator (the validator is not listed
+        in the validator registry)
+        """
```

Comment:
The comment is misleading.  The test is basically repeating the first test but this time instead of returning `None`, the validator registry is raising a `KeyError`.  Technically, the `KeyError` is the proper way for the validator registry to indicate a non-existent validator ID.  I would say remove the first test and prefer this test.